### ISSUE TITLE
[Feature Request] Is there an official tutorial to use Sockeye 2

### ISSUE THREAD

**Issue by sianvolta** [awslabs/sockeye#910]
As reported on https://www.aclweb.org/anthology/2020.eamt-1.50/ and https://arxiv.org/pdf/2008.04885.pdf, Sockeye 2 came with awesome features like 

 - Plateau Reduce (optimizer's learning rate tricks)
 - 8-bit Integer Quantization (Is there a post model training quantization binary?)
 - Source Factored models (hacks to the BPE processes and token embeddings)
 - AMP mixed precision training (FP16 and FP32 mixed precision from https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html)

Are there / Will there be tutorials on https://awslabs.github.io/sockeye/tutorials.html that shows how each/all of the above features can be used in the model training?

**Comment by tdomhan** [comment]
While there is no Sockeye 2 tutorial there is some documentation for individual features:
* Plateau reduce: this is on by default (see arguments `--learning-rate-reduce-num-not-improved` and `--learning-rate-reduce-factor`.
* Quantization: All you need to do is specify `--dtype int8` at inference time.
* Source factors: You can find some documentation on source factors here: https://awslabs.github.io/sockeye/training.html
* AMP: You can use the `--amp` argument in combination with Horovod `--horovod`. @mjdenkowski  anything else you'd like to point out in terms of horovod documentation?

**Comment by mjdenkowski** [comment]
Hi @sianvolta,

You can use the following settings to train a model similar to the one described in our [AMTA 2020 paper](https://arxiv.org/pdf/2008.04885.pdf) (20:2 transformer trained with large batches, AMP, and Horovod):

Preparing data:
```
python3 -m sockeye.prepare_data \
...
--bucket-width 8 \
--max-seq-len 96 \
--shared-vocab \
--pad-vocab-to-multiple-of 8
```

Model training:
```
horovodrun -np 8 python3 -m sockeye.train
...
--encoder transformer \
--decoder transformer \
--num-layers 20:2 \
--num-embed 512 \
--transformer-model-size 512 \
--transformer-attention-heads 8 \
--transformer-feed-forward-num-hidden 2048 \
--transformer-preprocess n \
--transformer-postprocess dr \
--weight-tying-type src_trg_softmax \
--learning-rate-scheduler-type plateau-reduce \
--learning-rate-reduce-num-not-improved 8 \
--learning-rate-reduce-factor 0.9 \
--batch-type max-word \
--batch-size 2560 \
--update-interval 16 \
--learning-rate-warmup 2000 \
--initial-learning-rate 0.00113 \
--checkpoint-interval 125 \
--checkpoint-improvement-threshold 0.001 \
--max-num-checkpoint-not-improved 60 \
--optimizer adam \
--loss cross-entropy \
--batch-sentences-multiple-of 8 \
--amp \
--horovod
```

See the [Dockerfile README](https://github.com/awslabs/sockeye/blob/master/sockeye_contrib/docker/README.md) for more information on running multi-process training.

**Comment by sianvolta** [comment]
Thank you @mjdenkowski and @tdomhan !! That instructions works well!

**Comment by fhieber** [comment]
Closing for now, feel free to reopen if you have more questions about this.
